include/storage/shared_datatype.hpp

```diff
@@ -157,7 +184,64 @@ class DataLayout
     }
 
     static constexpr std::size_t BLOCK_ALIGNMENT = 64;
-    std::map<std::string, Block> blocks;
+};
+
+class TarDataLayout final : public BaseDataLayout
+{
+  public:
+    inline void SetBlock(const std::string &name, Block block) override final
```

Comment:
These functions seem to do more or less the same across data layout child classes. Do you think there should be a common base impl. so we don't have to repeat in child classes?